functional-tests/fn_bridge_reassignment.py

```diff
@@ -0,0 +1,107 @@
+    After the dispatch assignment duration is over,
+    Check if new operator is being assigned or not
+    """
+
+    def __init__(self, ctx: flexitest.InitContext):
+        # Example: we spin up 5 operators
+        ctx.set_env(testenv.BasicEnvConfig(101, n_operators=3, pre_fund_addrs=True))
+
+    def main(self, ctx: flexitest.RunContext):
+        address = ctx.env.gen_ext_btc_address()
+        withdraw_address = ctx.env.gen_ext_btc_address()
+        self.debug(f"Address: {address}")
+        self.debug(f"Change Address: {withdraw_address}")
+
+        btc, seq, reth, seqrpc, btcrpc, rethrpc, web3, el_address = setup_services(ctx, self.debug)
+        btc_url = btcrpc.base_url
+        btc_user = btc.props["rpc_user"]
+        btc_password = btc.props["rpc_password"]
+        original_balance = get_balance(withdraw_address, btc_url, btc_user, btc_password)
+        self.debug(f"BTC balance before withdraw: {original_balance}")
+
+        # Check initial balance is 0
+        check_initial_eth_balance(rethrpc, el_address, self.debug)
+
+        # Perform two deposits
+        deposit_twice(ctx, web3, seqrpc, rethrpc, el_address, self.debug)
+
+        # Perform a withdrawal
+        l2_tx_hash, tx_receipt, total_gas_used = do_withdrawal(
+            web3, rethrpc, el_address, withdraw_address, self.debug, DEPOSIT_AMOUNT
+        )
+        original_balance = get_balance(withdraw_address, btc_url, btc_user, btc_password)
+        self.debug(f"BTC balance after withdraw: {original_balance}")
+
+        # Check assigned operator
+        duties = seqrpc.strata_getBridgeDuties(0, 0)["duties"]
+        withdraw_duty = [d for d in duties if d["type"] == "FulfillWithdrawal"][0]
+        assigned_op_idx = withdraw_duty["payload"]["assigned_operator_idx"]
+        assigned_operator = ctx.get_service(f"bridge.{assigned_op_idx}")
+        self.debug(f"Assigned operator index: {assigned_op_idx}")
+
+        # Stop assigned operator
+        self.debug("Stopping assigned operator ...")
+        assigned_operator.stop()
+
+        # Let enough blocks pass so the assignment times out
+        # (64 is the dispatch_assignment_duration)
+        btcrpc.proxy.generatetoaddress(64, UNSPENDABLE_ADDRESS)
+        time.sleep(3)
+
+        # Re-check duties
+        def my_val():
+            btcrpc.proxy.generatetoaddress(64, UNSPENDABLE_ADDRESS)
+            duties = seqrpc.strata_getBridgeDuties(0, 0)["duties"]
+            withdraw_duty = [d for d in duties if d["type"] == "FulfillWithdrawal"][0]
+            new_assigned_op_idx = withdraw_duty["payload"]["assigned_operator_idx"]
+            new_assigned_operator = ctx.get_service(f"bridge.{new_assigned_op_idx}")
+            self.debug(f"new assigned operator is {new_assigned_op_idx}")
+            return new_assigned_operator
+
+        new_assigned_operator = wait_until_with_value(
+            my_val,
+            predicate=lambda v: v != assigned_operator,
+            timeout=10,
+        )
```

Comment:
I think this check can be removed now since the reassignment logic is supposed to always compute a different assignee than the current one (by computing offset).